Update ProjectStructureAndRoles.md - Add ICM WG (#157)
camaraproject/Governance: ProjectStructureAndRoles.md

**Commonalities**

CAMARA commonalities shall be processed in a Working Group "Commonalities”. The Working Group “Commonalities” works under the supervision of the TSC. The scope of the Working Group is to create guidelines, best practices and concept documents relevant for all APIs in CAMARA. The Commonalities Working Group shall work e.g. on the development of a common meta model that shall be used as a reference across the Sub Projects. It is open for members of any Sub Project or Working Group to raise issues in the Technical Steering Committee (TSC) for relevant topics on a need basis.

> All Sub Projects must comply with the work in the Commonalities Working Group.

Reviewer: I don't believe there is any obligation to comply with updates that are not yet part of a release
```suggestion
All Sub Projects must comply with the latest release of the Commonalities Working Group guidelines.
```

Reviewer 2: Somehow this PR pivoted in scope from adding ICM (which is not yet done) to correcting the Commonalities section. But as we will get the same sentence also for ICM it makes sense to discuss.

@eric-murray I don't like to reduce it just to "latest releases". During the release cycle there are for examples also pre-releases relevant which are excluded by the term "latest release" (at least in the sense GitHub is using it). 

How about just changing "work" into "work results"?

Reviewer: Maybe we can review the text after the meta-release cycle is fully established and sub-project responsibilities clear.

I think sub-projects need clearer guidance on the specific Commonalities (and ICM) rules and guidelines they need to comply with at a given time. It is not unknown for changes to be made to the WIP documentation, only for those changes to be later modified before being "frozen" in a release. It is hard for sub-projects to try and hit a moving target.

I include pre-releases (release candidates) here, as I don't think it should be mandatory to comply even with a release candidate, only the release itself, though I understand the timescales for the current meta-release force us to do this at the moment.

Reviewer 2: Yes, would be good to have this discussion independently. I agree that the current sentence is vague, but would like to keep it for now.

**Commonalities**

> CAMARA commonalities shall be processed in a Working Group "Commonalities”. The Working Group “Commonalities” works under the supervision of the TSC. The scope of the Working Group is to create guidelines, best practices and concept documents relevant for all APIs in CAMARA. The commonalities Working Group shall work e.g. on the development of a common meta model that shall be used as a reference across the Sub Projects. It is open for members of any Sub Project or Working Group to raise issues in the Technical Steering Committee (TSC) for relevant topics on a need basis.

Reviewer 3: ```suggestion
CAMARA commonalities shall be processed in a Working Group "Commonalities”. The Working Group “Commonalities” works under the supervision of the TSC. The scope of the Working Group is to create guidelines, best practices and concept documents relevant for all APIs in CAMARA. The Commonalities Working Group shall work e.g. on the development of a common meta model that shall be used as a reference across the Sub Projects. It is open for members of any Sub Project or Working Group to raise issues in the Technical Steering Committee (TSC) for relevant topics on a need basis.
```

All Sub Projects must comply with the work in the commonalities Working Group.

**Identity and Consent Management**

In the Identity and Consent Management Working Group solutions are defined and developed to identify users/subscribers, to capture, store and manage user consent, and to make sure that end user experience of using the API is not compromised.

> 

Reviewer 4: I assume that the Working Group “Identity and Consent Management” (as Commonalities) also works under the supervision of the TSC.

On the other hand, as Commonalities, all subprojects must be consistent (comply) with the work of the Identity and Consent Management Working Group.

Finally, we can also mention the ICM work around API access definitions and the Security and Interoperability Profile, which includes CAMARA definitions for AuthN/AuthZ flows.

These are just suggestions :)

> **Identity and Consent Management**
> 
> In the Identity and Consent Management Working Group solutions are defined and developed to identify users/subscribers, to capture, store and manage user consent, and to make sure that end user experience of using the API is not compromised.
> 

Reviewer 4: ```suggestion
**Identity and Consent Management**

The Identity and Consent Management Working Group, operating under the supervision of the Technical Steering Committee (TSC), focuses on defining and developing comprehensive solutions for user/subscriber identification, consent capture, storage, and management. These solutions ensure that user experience with the API remains seamless and uncompromised.

All Sub Projects must comply with the work in the "Identity and Consent Management" Working Group. The Working Group's efforts include defining API access parameters and developing the Security and Interoperability Profile, which encompasses CAMARA definitions for authentication and authorization (AuthN/AuthZ) flows.
```
Does this look good? Please feel free to change anything you like.

Reviewer 2: Fair point to express the same level of governance and compliance for Identity and Consent Management.

Small changes in my proposal below:
* I haven't understood what is meant with "defining API access parameters" so I left it out. 
* Proposal to not start a new explanation after the sentence "All Sub Projects must comply with the work in the Identity and Consent Management Working Group.
* replaced "uncompromised" (which my syntax check does not like) with "protected".

```suggestion
**Identity and Consent Management**
The Identity and Consent Management Working Group, operating under the supervision of the Technical Steering Committee (TSC), focuses on defining and developing comprehensive solutions for user/subscriber identification, consent capture, storage, and management. These solutions ensure that user experience with the API remains seamless and protected. 

The Working Group's efforts include developing the Security and Interoperability Profile, which encompasses CAMARA definitions for authentication and authorization (AuthN/AuthZ) flows.

All Sub Projects must comply with the work in the Identity and Consent Management Working Group.
```

Reviewer 3: I do not want it to be too complicated but usually there is no "user identification".
Telcos do identify the subscriber but we do not identify the user. At least not what commonly is understood as the "user".

Also, I would rephrase making it clearer that "storage, and management" are related to consent.

> The Identity and Consent Management Working Group, operating under the supervision of the Technical Steering Committee (TSC), focuses on defining and developing privacy-first solutions for subscriber identification, and user authentication, and the purpose of the API access, and user consent capture, storage, and management. These solutions ensure that the API access is always authorized and secure. 

That got somewhat lengthy...
